curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data.english.md

```diff
@@ -7,10 +7,45 @@ forumTopicId: 301313
 
 ## Description
 <section id='description'>
+
 <code>Array.prototype.reduce()</code>, or simply <code>reduce()</code>, is the most general of all array operations in JavaScript. You can solve almost any array processing problem using the <code>reduce</code> method.
-This is not the case with the <code>filter</code> and <code>map</code> methods since they do not allow interaction between two different elements of the array. For example, if you want to compare elements of the array, or add them together, <code>filter</code> or <code>map</code> could not process that.
+
 The <code>reduce</code> method allows for more general forms of array processing, and it's possible to show that both <code>filter</code> and <code>map</code> can be derived as a special application of <code>reduce</code>.
-However, before we get there, let's practice using <code>reduce</code> first.
+The <code>reduce</code> method iterates over each item in an array and returns a single value (i.e. string, number, object, array).  This is acheived via a callback function that returns a value created by the user's logic after each iteration.
+
+The callback function accepts four arguments (with names of your choice).  While iterating over the array, the arguments can be referenced in the callback function.  The first argument is known as the accumulator, which gets assigned the return value of the callback function during each iteration.  The second argument is the current element being processed.  The third argument is the index of the current element being processed.  The fourth argument is the actual array upon which the <code>reduce</code> method was called.
+
+In addition to the callback function, the <code>reduce</code>method has an additional parameter which takes an initial value for the accumulator.  If this second parameter is not used, then the accumulator's initial value is the first element of the array and the second argument is the second element in the first iteration.
+
+See below for an example using the <code>reduce</code> method on a <code>users</code> array containing objects to return the sum of all the users' ages. The example only uses the first and second arguments, but there are times when you may need to reference the third and/or fourth arguments.
```

Comment:
```suggestion
See below for an example using <code>reduce</code> on the <code>users</code> array to return the sum of all the users' ages. For simplicity the example only uses the first and second arguments.
```

Reply:
Can you add a comma after `For simplicity` so I can commit this?